modules/assim.sequential/R/build_X.R

```diff
@@ -0,0 +1,94 @@
+
+  if(t == 1){
+    reads <-
+      furrr::future_pmap(list(out.configs %>% `class<-`(c("list")), settings, new.params),function(configs,settings,siteparams) {
+        # Loading the model package - this is required bc of the furrr
+        #library(paste0("PEcAn.",settings$model$type), character.only = TRUE)
+        #source("~/pecan/models/sipnet/R/read_restart.SIPNET.R")
+        
+        X_tmp <- vector("list", 2)
+        
+        for (i in seq_len(nens)) {
+          X_tmp[[i]] <- do.call( my.read_restart,
+                                 args = list(
+                                   outdir = outdir,
+                                   runid = settings$runs$id[i] %>% as.character(),
+                                   stop.time = read_restart_times[t+1],
+                                   settings = settings,
+                                   var.names = var.names,
+                                   params = siteparams[[i]]
+                                 )
+          )
+          
+        }
+        return(X_tmp)
+      })
+    
+  }else{
+    reads <-
+      furrr::future_pmap(list(out.configs %>% `class<-`(c("list")), settings, new.params),function(configs,settings,siteparams) {
+        # Loading the model package - this is required bc of the furrr
+        #library(paste0("PEcAn.",settings$model$type), character.only = TRUE)
+        #source("~/pecan/models/sipnet/R/read_restart.SIPNET.R")
+        
+        X_tmp <- vector("list", 2)
+        
+        for (i in seq_len(nens)) {
+          X_tmp[[i]] <- do.call( my.read_restart,
+                                 args = list(
+                                   outdir = outdir,
+                                   runid = configs$runs$id[i] %>% as.character(),
+                                   stop.time = read_restart_times[t+1],
+                                   settings = settings,
+                                   var.names = var.names,
+                                   params = siteparams[[i]]
+                                 )
+          )
+          
+        }
+        return(X_tmp)
+      })
+  }
+  
+  # #let's read the parameters of each site/ens
+  # params.list <- reads %>% map(~.x %>% map("params"))
+  # # Now let's read the state variables of site/ens
+  # X <- reads %>% map(~.x %>% map_df(~.x[["X"]] %>% t %>% as.data.frame))
+  # 
+  # # Now we have a matrix that columns are state variables and rows are ensembles.
+  # # this matrix looks like this
+  # #         GWBI    AbvGrndWood   GWBI    AbvGrndWood
+  # #[1,]  3.872521     37.2581  3.872521     37.2581
+  # # But therer is an attribute called `Site` which tells yout what column is for what site id - check out attr (X,"Site")
+  # if (multi.site.flag){
+  #   X <- X %>%
+  #     map_dfc(~.x) %>% 
+  #     as.matrix() %>%
+  #     `colnames<-`(c(rep(var.names, length(X)))) %>%
+  #     `attr<-`('Site',c(rep(site.ids, each=length(var.names))))
+  # }
```

Comment:
This comment block looks like it may be good documentation -- material should it move to the @examples section or maybe to a vignette?

Reply:
Yes, I wasn't sure where to put this. While I created this function I did not write this code, the code used to exist within the main sda.enkf_MultiSite.R function. So when I created the function I did not want to get rid of the previous documentation. This function is very much a work in progress.